### ISSUE TITLE
intake server documentation

### ISSUE THREAD

**Issue by delgadom** [ContinuumIO/intake#198]
The documentation references the ability to set up an intake server, but does not point to any resources about how to set this up. do docs on server setup exist somewhere?

**Comment by martindurant** [comment]
The documentation is [here](https://intake.readthedocs.io/en/latest/tools.html#intake-server), perhaps a little hidden and certainly could be expanded. 

Probably the protocol itself should be described in detail, though a glance at [the implementation](https://github.com/ContinuumIO/intake/blob/master/intake/cli/server/server.py#L20) shows it is pretty simple. As I have been telling people, the provided server could be seen as a prototype/example, and if the concept proves useful, I expect more server implementations to appear.

**Comment by martindurant** [comment]
@delgadom , I am interested, though: what is your use case?

**Comment by delgadom** [comment]
Oh thank you @martindurant! Yes I had completely missed this in reading through the docs. Maybe some internal links from other pages would be helpful, if not a whole section (not sure if you think this warrants this).

Still getting spun up on this (evaluating using this for our research team), but are there types of storage formats available to the server besides yml? Where can one find information about server configuration? Is a source dive necessary here?

Also if there are any example deployments you think are particularly helpful a link would be great. E.g. is there a pangeo deployment that could serve as a particularly useful intake server deployment?

I'd be happy to add to the docs as I work through these questions if it would be helpful.

**Comment by delgadom** [comment]
Re: our use case, we have two pangeo-style kubernetes clusters with 10 or so active users. We split off of the core pangeo development process because we needed some core features really early on that weren't supported and have been too busy to rebase to the current deployments, but intend to rejoin the flock if possible soon. Looking to see if intake would be a good data catalogue & versioning system for that deployment, and perhaps also for our other computing resources (30 active users spread across the Berkeley high performance computing center, a standalone Linux server at Berkeley, and some nodes at the open science data cloud at UChicago). We do climate economic impact research, so our data is pretty typical geoscience data with 1000s of frequently accessed netcdf, zarr, CSV, parquet files, along with other less standard file types (e.g. stats .dat and Matlab .mat files).

**Comment by martindurant** [comment]
Agree, that it should have its own section and certainly better internal linking. For exampe, "Intake Server" appears in the glossary, but doesn't include the link I gave here.

> are there types of storage formats available to the server besides yml

Internally, the server calls `intake.open_catalog` with the CLI argument, which can be a single YAML file, a YAML glob or the URL of another intake server. The function called takes an optional `driver=` parameter, and there is absolutely no reason why it shouldn't be some other type of catalogue such as SQL tables. It maybe would take a little thinking about the right way to express this on the command line or configuration.

> is there a pangeo deployment that could serve as a particularly useful intake server

Actually, pangeo uses a [flat YAML file](https://github.com/pangeo-data/pangeo/raw/master/gce/catalog.yaml) (with other files for the other pangeo deployments). Their only extra is to include that address as the default lookup in the config, parameter `'catalog_path'`.

**Comment by martindurant** [comment]
Your use case sounds fun! You *might* want to indeed use the server to provide up-to-date data specs and maybe even stream data that is inconvenient for cloud use, i.e., hdf5. You could also follow the policy of main pangeo and have a flat file in the repo which you reference; in their case, they use FUSE to allow access to hdf5 files as if they were local of opendap if appropriate, but that may come at a performance penalty. Also, it is possible to use Intake's caching mechanism to download files on first access, if the machines are to have enough local storage for that.

So... lots of options!
